common/configurable/src/main/java/io/helidon/common/configurable/ThreadPool.java

```diff
@@ -131,18 +239,603 @@ public int getQueueCapacity() {
      * @return The size.
      */
     public int getPeakQueueSize() {
-        return peakQueueSize.get();
+        return queue.getPeakSize();
+    }
+
+    /**
+     * Returns the number of completed tasks.
+     *
+     * @return The count.
+     */
+    public int getCompletedTasks() {
+        return completedTasks.get();
+    }
+
+    /**
+     * Returns the number of tasks that threw an exception.
+     *
+     * @return The count.
+     */
+    public int getFailedTasks() {
+        return failedTasks.get();
+    }
+
+    /**
+     * Returns the number of completed and failed tasks.
+     *
+     * @return The count.
+     */
+    public int getTotalTasks() {
+        return completedTasks.get() + failedTasks.get();
+    }
+
+    /**
+     * Returns the current number of active threads.
+     *
+     * @return The count.
+     */
+    public int getActiveThreads() {
+        return activeThreads.get();
+    }
+
+    /**
+     * Returns the average number of active threads across the life of the pool.
+     *
+     * @return The average.
+     */
+    public float getAverageActiveThreads() {
+        final float totalActive = totalActiveThreads.sum();
+        if (totalActive == 0) {
+            return 0.0f;
+        } else {
+            return totalActive / (float) getTotalTasks();
+        }
+    }
+
+    /**
+     * Returns the rejection count.
+     *
+     * @return The count.
+     */
+    public int getRejectionCount() {
+        return rejectionPolicy.getRejectionCount();
+    }
+
+    /**
+     * Tests whether or not the number of threads can change over time.
+     *
+     * @return {@code true} if maximum size is equal to core size.
+     */
+    public boolean isFixedSize() {
+        return getMaximumPoolSize() == getCorePoolSize();
+    }
+
+    @Override
+    public WorkQueue getQueue() {
+        return queue;
+    }
+
+    /**
+     * Returns the current number of tasks in the queue.
+     *
+     * @return The count.
+     */
+    public int getQueueSize() {
+        return queue.size();
+    }
+
+    @Override
+    public void setRejectedExecutionHandler(RejectedExecutionHandler handler) {
+        if (handler instanceof RejectionPolicy) {
+            super.setRejectedExecutionHandler(handler);
+        } else {
+            throw new IllegalArgumentException(handler.getClass() + " must be an instance of " + RejectionPolicy.class);
+        }
+    }
+
+    @Override
+    public void setMaximumPoolSize(int maximumPoolSize) {
+        throw new UnsupportedOperationException();
```

Comment:
This is a backward incompatible change.

Reply:
True. Given that setting this setting is generally pointless anyway, even without my changes, I guess making it a No-Op is a better choice, perhaps with a log message.

Reply:
Changed this to log a warning.